lib/active_model/serializer/reflection.rb

```diff
@@ -79,6 +91,20 @@ def build_association(subject, parent_serializer_options)
 
       private
 
+      def condition_type
+        if options.key?(:if)
+          :if
+        elsif options.key?(:unless)
+          :unless
+        else
+          :none
+        end
+      end
+
+      def condition
+        options[condition_type]
+      end
```

Comment:
Agreed, we could either make a `Field` class from which `Attribute` and `Reflection` would inheritate, or a `ConditionalField` mixin. Not sure what's best, and naming is obviously up for debate.